docs/ELK-Stack.md

```diff
@@ -0,0 +1,368 @@
+
+Logstash defines pipeline as well. In the pipeline configuration of Logstash you need to specify where it should look for IntelMQ events, what to do with them and where to pass them.
+
+### Input
+
+This part describes how to receive data from Redis queue.
+See the example configuration and comments below:
+
+```
+input {
+  redis {
+    host => "10.10.10.10"
+    port => 6379
+    db => 2 
+    data_type => "list"
+    key => "restapi-output-queue"
+  }
+}
+```
+* `host` - Redis hostname or IP
+* `port` - Redis port
+* `db` - Redis db used (same value as `source_pipeline_db` from defaults.conf of IntelMQ)
+* `data_type` - data type is usually `list`
+* `key` - name of the queue (found in pipeline.conf of IntelMQ as source queue for our dummy bot)
+
+**Extra**
+
+You can also use syntax like this: `host => "${REDIS_HOST:10.10.10.10}"`
+
+The value will then be taken from environment variable `$REDIS_HOST`. If the environment variable is not defined then the default value of `10.10.10.10` will be used instead.
+
+### Filter (optional)
+
+Before passing the data to the database you can apply certain changes. This is done with filters. See an example: 
+```
+filter {
+  mutate {
+    lowercase => ["source.geolocation.city", "classification.identifier"]
+    remove_field => ["__type", "@version"]
+  }
+  date {
+    match => ["time.observation", "ISO8601"]
+  }
+}
+```
+I recommend using the `date` filter: generally we have two timestamp fields - `time.source` (provided by feed source this can be understood as when the event happend; however it is not always present) and `time.observation` (when IntelMQ collected this event). Logstash also adds another field `@timestamp` with time of processing by Logstash. While it can be useful for debugging, I recommend to set the `@timestamp` to the same value as `time.observation`.
+
+### Output 
+
+The pipeline also needs output, where we define our database (Elasticsearch). The simplest way of doing so is defining an output like this:
+```
+output {
+  elasticsearch {
+    hosts => ["http://10.10.10.11:9200", "http://10.10.10.12:9200"]
+    index => "intelmq-%{+YYYY.MM}"
+  }
+}
+```
+* `hosts` - Elasticsearch host (or more) with the correct port (9200 by default)
+* `index` - name of the index where to insert data
+
+Our experience, hardware equipment and the amount of events collected led us to having a separate index for each month. This might not necessarily suit your needs.
+
+## Configuring Elasticsearch
+
+Configuring Elasticsearch is entirely up to you and should be consulted with the [official documentation](https://www.elastic.co/guide/en/elasticsearch/reference/index.html). What you will most likely need is something called [index template](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-templates.html) mappings which I can provide. Again it might not suit your needs so feel free to adjust it for your use case, it is only included as an inspiration.
+
+```
+{
```

Comment:
We have an es mapper tool which should be used here.

Reply:
This is excellent, thanks for pointing it out, I will update the guide accordingly.